Net01_2/DiagonalMatrix.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+
+namespace Net01_2
+{
+    internal class DiagonalMatrix<T> : SquareMatrix<T>
+    {
+        /// <summary>
+        /// Sets size of matrix.
+        /// </summary>
+        /// <param name="size">The size.</param>
+        /// <exception cref="System.ArgumentException">
+        /// <paramref name="size" /> must be non-negative and not equal to zero.
```

Comment:
You allocate memory twice, in basic class ctor, and in this ctor. Try to fix, only one allocation has to be